app/src/screens/Onboarding/SignInScreen.tsx

```diff
@@ -13,8 +39,69 @@ import { StackScreenProps } from '@react-navigation/stack';
 type SignInScreenProps = StackScreenProps<OnboardingStackParams, 'SignInScreen'>;
 
 export const SignInScreen = ({ navigation }: SignInScreenProps) => {
+  // Determines if text in password input is visible or not
+  const [secureTextEntry, setSecureTextEntry] = useState<boolean>(true);
+
+  // Changes passwordInputVisible Value for when eye icon is passed
+  const TogglePasswordVisibility = () => {
+    if (!secureTextEntry) {
+      setSecureTextEntry(true);
+    } else {
+      setSecureTextEntry(false);
+    }
+  };
+
   return (
-    <FullScreenWhite>
+    <SignInMainContainer>
+      <SignInHeaderContainer>
+        <LogoContainer />
+        <SignInHeaderTextContainer>
+          <SignInHeaderTitleText>Universal Store</SignInHeaderTitleText>
+          <SignInSubHeaderText>Redefining express checkout.</SignInSubHeaderText>
+        </SignInHeaderTextContainer>
+      </SignInHeaderContainer>
+
+      <SignInHeaderTitleText>Sign In</SignInHeaderTitleText>
+      <SignInSubHeaderText>Welcome back!</SignInSubHeaderText>
+
+      <SignInFormContainer>
+        <SignInFormText>Email</SignInFormText>
+
+        <SignInInputContainer>
+          <SignInTextInputIconContainer>
+            <EmailIcon></EmailIcon>
+          </SignInTextInputIconContainer>
+          <SignInTextInput></SignInTextInput>
+        </SignInInputContainer>
+
+        <SignInFormText>Password</SignInFormText>
+
+        <SignInInputContainer>
+          <SignInTextInputIconContainer>
+            <LockIcon></LockIcon>
+          </SignInTextInputIconContainer>
+          <SignInTextInput secureTextEntry={secureTextEntry ? true : false}></SignInTextInput>
+          <SignInVisibleIconButton onPress={TogglePasswordVisibility}>
```

Comment:
instead of calling this function, call () => setSecureTextEntry(!secureTextEntry)